src/server.h

```diff
@@ -385,6 +385,9 @@ extern int configOOMScoreAdjValuesDefaults[CONFIG_OOM_COUNT];
                                       memory eviction. */
 #define CLIENT_ALLOW_OOM (1ULL<<44) /* Client used by RM_Call is allowed to fully execute
                                        scripts even when in OOM */
+#define CLIENT_CUSTOM_AUTH (1ULL<<45) /* Indicates the client has module based auth in progress */
```

Comment:
This flag "implies" it's blocked. I wonder if instead of this we can just check if it's blocked and has the custom handler set.

Reply:
No, this flag does not imply that the client is blocked, it just means that the client is attempting custom authentication. This flag is used in multiple places - (1) to check if any custom auth is ongoing and to return ERR during module unloads if so  (2) to allow auth chains to exist by avoiding resetting the client args between blocking auth callbacks (3) to queue the client for reprocessing if custom auth has not concluded after one cycle of a blocking auth callback (in this case the next cb will be attempted) (4) to assert that the RM_BlockClientOnAuth can be used because the client is in the context of custom auth

There are other cases where this is needed as well

Reply:
I think the confusion is that clients only retain some authentication state when they are blocked, in all other cases all the state is retired. Just do go through your points:

1) This can only happen in the context of blocking, since in all other cases the client will immediately give up the state.
2) That is for blocking commands right? So my point seems appropriate here?
3) This is also in the context of blocking.
4) This is also in the context of blocking.

Generally speaking I don't want flags when there is other state that directly reflects, which I think in this case is blocking context.

Reply:
I replaced the `CLIENT_CUSTOM_AUTH` client flag with the `isClientModuleAuthInProgress` Macro in this commit - https://github.com/redis/redis/pull/11659/commits/e0500097e976305e654fb1fac53e03251c9da288